third_party/mbedtls/mbedtls-config.h

```diff
@@ -699,6 +708,32 @@ __inline int windows_kernel_snprintf(char * s, size_t n, const char * format, ..
  */
 //#define MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED
 
+#if OPENTHREAD_ENABLE_APPLICATION_COAP_SECURE
+/**
+ * Enables ciphersuite ECDHE ECDSA with x.509 certificate
+ * for Application Coap Secure.
+ */
+#ifndef MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED
+//#define MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED
+#endif // MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED
+#if defined(MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED)
+
+/**
+ * to increase the security with ECDHE ECDSA enable
+ * ECDSA DETERMINISTIC and X509 CHECK KEY USAGE.
+ * Not required!
+ */
+#ifndef MBEDTLS_ECDSA_DETERMINISTIC
+//#define MBEDTLS_ECDSA_DETERMINISTIC
+#endif // MBEDTLS_ECDSA_DETERMINISTIC
+
+#ifndef MBEDTLS_X509_CHECK_KEY_USAGE
+//#define MBEDTLS_X509_CHECK_KEY_USAGE
```

Comment:
Should this be enabled when `MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED` is `1`?

Reply:
Yes, it should be (but must not) enabled, when `MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED` is defined.

Reply:
Recognized.